scanner/matcher/updater/vuln/dist.go

```diff
@@ -0,0 +1,55 @@
+package vuln
+
+import (
+	"context"
+	"slices"
+
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/scanner/datastore/postgres"
+)
+
+// distManager manages updates to known-distributions.
+type distManager struct {
+	store postgres.MatcherStore
+
+	// mutex protects access to known.
+	mutex sync.RWMutex
+	// known tracks all known distributions.
+	known []claircore.Distribution
+}
+
+func newDistManager(store postgres.MatcherStore) *distManager {
+	return &distManager{
+		store: store,
+	}
+}
+
+// update updates the currently known distributions, on-demand.
+//
+// It is *not* safe to call update concurrently.
```

Comment:
I could make this safe with a `singleflight` group https://pkg.go.dev/golang.org/x/sync/singleflight#Group

The point is: we should not hold the mutex while reaching out to the DB, yet we want `update` to be atomic. Currently, `update` is only called synchronously, so we can accept this risk. If we want to be safe, though, I can easily add a `singleflight` group to make `update` atomic. Internally, `singleflight` adds duplicate function calls to a `sync.WaitGroup`